Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -9015,6 +9015,272 @@ def inner(client, args):
     return dec
 
 
+def get_fetch_run_time_range(last_run, first_fetch, look_back=0, timezone=0, date_format='%Y-%m-%dT%H:%M:%S'):
+    """
+    Gets the time range for fetch
+
+    :type last_run: ``dict``
+    :param last_run: The LastRun object
+
+    :type first_fetch: ``str``
+    :param first_fetch: The first time to fetch, used in the first fetch
+
+    :type look_back: ``int``
+    :param look_back: The time to look back in fetch in minutes
+
+    :type timezone: ``int``
+    :param timezone: The time zone offset in hours
+
+    :type date_format: ``str``
+    :param date_format: The date format
+
+    :return: The time range of the creation date for the incidents to fetch in the current run.
+    :rtype: ``Tuple``
+    """
+    last_run_time = last_run and 'time' in last_run and last_run['time']
+    now = datetime.utcnow() + timedelta(hours=timezone)
+    if not last_run_time:
+        last_run_time = dateparser.parse(first_fetch, settings={'TIMEZONE': 'UTC'}) + timedelta(hours=timezone)
+    else:
+        last_run_time = dateparser.parse(last_run_time, settings={'TIMEZONE': 'UTC'})
+
+    if look_back > 0:
+        if now - last_run_time < timedelta(minutes=look_back):
+            last_run_time = now - timedelta(minutes=look_back)
+
+    return last_run_time.strftime(date_format), now.strftime(date_format)
+
+
+def filter_incidents_by_duplicates_and_limit(incidents_res, last_run, fetch_limit, id_field='id'):
```

Comment:
`fetch_limit` should have a default value of `-1` that will mean "no fetch limit" and will return all filtered incidents.

Reply:
basically we prefer to always put a default value to a fetch limit param to avoid performance issues. perhaps instead of `-1` we can do `200` as a default?

Reply:
The incidents were already fetched,
so there are no performance issues related to remote (e.g. network latency), or local (e.g. memory)

Reply:
the issue can be during the execution of the fetched incidents, AFAIK. @anara123 can you advise?